code_samples/logging.js

```diff
@@ -0,0 +1,81 @@
+var winston = require('winston');
+var Config = require('../.').Config;
+var Controller = require('../test/RC');
+var HazelcastClient = require('../.').Client;
+
+
+var winstonAdapter = {
+    logger: new (winston.Logger)({
+        transports: [
+            new (winston.transports.Console)()
+        ]
+    }),
+
+    levels: [
+        'error',
+        'warn',
+        'info',
+        'debug',
+        'silly'
+    ],
+
+    log: function(level, className, message, furtherInfo) {
+        this.logger.log(this.levels[level], className + ' ' + message);
+    }
+};
+
+var loggingHappened = false;
+
+
+winstonAdapter.logger.on('logging', function(transport, level, msg, meta) {     
+    loggingHappened = true;
+});
+
+//winston logging is active in case of winston object given as property
+var cfg = new Config.ClientConfig();
+cfg.properties['hazelcast.logging'] = winstonAdapter;
+HazelcastClient.newHazelcastClient(cfg)
+    .then(function(client){
+        return client.shutdown();
+    })
+    .then(function(){
+        //logging is not active in case of off property
+        cfg.properties['hazelcast.logging'] = 'off';
+        return HazelcastClient.newHazelcastClient(cfg);
+    })
+    .then(function(client){
+        return client.shutdown();
+    })
+    .then(function(){
+        //default logging is active in case of default property
+        cfg.properties['hazelcast.logging'] = 'default';
+        return HazelcastClient.newHazelcastClient(cfg);
+    })
+    .then(function(client){
+        return client.shutdown();
+    })
+    .then(function(){
+        //default logging is active in case of empty property
+        return HazelcastClient.newHazelcastClient(cfg);
+    })
+    .then(function(client){
+        return client.shutdown();
+    });
+
+//Error is thrown in case of unknown property
+cfg.properties['hazelcast.logging'] = 'asdas';
+HazelcastClient.newHazelcastClient(cfg)
```

Comment:
It looks like you open two client instances simultaneously. This one and the one with "winstonAdapter". Their logs may mix.